clientv3/watch.go

```diff
@@ -489,19 +497,31 @@ func (w *watchGrpcStream) dispatchEvent(pbresp *pb.WatchResponse) bool {
 	w.mu.RLock()
 	defer w.mu.RUnlock()
 	ws, ok := w.streams[pbresp.WatchId]
+	if !ok {
+		return false
+	}
+	if pbresp.Created {
+		// ignore created event if the watcher is not requested it
+		// or we already sent the initial created event (when we are on
+		// the resume path).
+		if !ws.initReq.createdNotify || ws.lastRev != 0 {
```

Comment:
On second thought, that will still queue up extra create events. If `serveStream` binds `lastRev` on the first `<-ws.recvc` when `createdNotify` is set, that should be OK.